### PR TITLE
ENH: chxtools package update for 2.0.x

### PR DESCRIPTION
**Pull request by mrakitin**
Here is the updated set of files for `chxtools`, which now contains the updated list of dependencies which are installed during `pip install`. There are some outdated files, which I didn't touch, e.g.:
- https://github.com/NSLS-II-CHX/chxtools/blob/enh_chxtools_pkg/chxtools/develop.py
- https://github.com/NSLS-II-CHX/chxtools/blob/enh_chxtools_pkg/chxtools/ophyd_tools.py
- https://github.com/NSLS-II-CHX/chxtools/blob/enh_chxtools_pkg/chxtools/plot.py

We should probably do something regarding `handlers.py` and `handlers2.py`. Either merge them, or use the ones from `eiger-io`. Ideas?

### CODE REVIEW

**Comment by mrakitin** [comment]
@yugangzhang, do you know how to fix this issue? That should be imported from somewhere.
```py
In [4]: import chxtools.chx_utilities
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-4-16a3c9880e42> in <module>()
----> 1 import chxtools.chx_utilities

/home/xf11id/Repos/chxtools/chxtools/chx_utilities.py in <module>()
   1472     gs.TABLE_COLS = [gs.PLOT_Y]
   1473 
-> 1474 def get_ID_calibration(gapstart,gapstop,xray_eye1=xray_eye1, gapstep=.2,gapoff=0,sl=300):
   1475     """
   1476     by LW 04/20/2015

NameError: name 'xray_eye1' is not defined
```
You can change to 
def get_ID_calibration(gapstart,gapstop,xray_eye1=None, gapstep=.2,gapoff=0,sl=300):

File: chxtools/handlers2.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import numpy as np
 import h5py
 from pims import FramesSequence, Frame
-from databroker import db
+from databroker import Broker
+from .handlers import LazyEigerHandler
```
Comment (jrmlhermitte) [inline chxtools/handlers2.py:5]: eventually, should be 
`from eiger_io.fs_handler import EigerHandler`
for `PIMS` version
or
`from eiger_io.fs_handler_dask import EigerDaskHandler`
for `dask` version.

But let's keep it as is for now. There are some things I need to check back with Yugang on.

Just to summarize what needs to be fixed with this:
1. The handlers that treat the `AD_EIGER` and `AD_EIGER2` specs should be combined into one. The only difference between the two is that the keyword argument is a different name. (Other that that API is same as far as I can tell and I've already tested this on all versions of EIGER files, of which there are four in CHX currently)
2. The `EigerImages` `pims` subclass (which is created by the handler per event) contains a lot of metadata which is obtained by reading from the hdf5 file directly. We should discourage that and try to make sure CHX accesses this data directly from `metadatastore`. Most of this data should be in a descriptor `configuration`. I want to discuss with Yugang and try my best to make sure we weed this out, and make a long term plan for deprecating this use.

File: chxtools/handlers2.py [hunk 2]
```diff
@@ -96,8 +97,11 @@ def __call__(self, seq_id):
...
         return EigerImages2(master_path, self._images_per_file, md=md)
 
-db.fs.deregister_handler('AD_EIGER')
-db.fs.deregister_handler('AD_EIGER2')
+db = Broker.named('chx')
+
+db.reg.deregister_handler('AD_EIGER')
+db.reg.deregister_handler('AD_EIGER2')
+
+db.reg.register_handler('AD_EIGER2', EigerHandler2)
```
Comment (jrmlhermitte) [inline chxtools/handlers2.py:105]: you could remove the previous two lines and add the `overwrite=True` flag to this line and bellow. Maybe we should get @danielballan 's blessing first ;-)

File: chxtools/speckle.py [hunk 3]
```diff
@@ -10,8 +10,8 @@
 import numpy as np
 import time
 
-from skxray.core import roi
-from skxray.core.utils import bin_edges_to_centers, geometric_series
+from skbeam.core import roi
```
Comment (jrmlhermitte) [inline chxtools/speckle.py:13]: for some history: `scikit-xray` was renamed to `scikit-beam` a while ago. So just substituting the name is perfect.

File: setup.py [hunk 4]
```diff
@@ -7,6 +7,13 @@
 
 from setuptools import setup, find_packages
 
+no_git_reqs = []
```
Comment (jrmlhermitte) [inline setup.py:10]: :+1:

**Review by jrmlhermitte — APPROVED** [review]
Looks good. I see lots of style changes too. Was the whole document checked for flake8?

We could consider adding a flake8 checker for Travis (no tests yet).

This PR only shows a small amount of what @mrakitin had to do to get this all working. Good job looks great! :-)

File: chxtools/handlers2.py (hunk 1 above)
Comment (jrmlhermitte) [reply chxtools/handlers2.py:5]: PR here: https://github.com/NSLS-II-CHX/eiger-io/pull/8

**Comment by jrmlhermitte** [comment]
For @mrakitin 's previous comment. Just noting that we understood that `xray_eye1` is an ophyd object assumed to be in the name space. This should be changed.

**Comment by jrmlhermitte** [comment]
one more comment : fix the readme if you can (there are three titles)

File: chxtools/handlers2.py (hunk 2 above)
Comment (danielballan) [reply chxtools/handlers2.py:105]: Blessed!

File: chxtools/chx_wrapper.py [hunk 5]
```diff
@@ -16,19 +17,19 @@ def det_select(det):
     calling sequence: det_select(det)
     """
     try:
-	rm_det=ascan.user_detectors[0].name
-	ascan.user_detectors.remove(session_mgr[rm_det])
+        rm_det=ascan.user_detectors[0].name
```
Comment (danielballan) [inline chxtools/chx_wrapper.py:20]: This is using _very old_ ophyd API that definitely no longer works. It took me a minute to even remember `.user_detectors`; I think it goes back to the days when the RunEngine was part of ophyd and there was no such thing as plans.  @mrakitin Can you open an issue to revisit this function?

File: chxtools/chx_wrapper.py [hunk 6]
```diff
@@ -43,30 +44,32 @@ def cw_ascan(mot,xmin,xmax,npoints,acqt='default',pos_ret=True):
     WILL NOT WORK FOR A LIST OF DETECTORS!
     """
     # gather beamline information prior to starting the scan:
-    ini_motpos=caget(mot.record+'.RBV')
+    ini_motpos = caget(mot.record+'.RBV')
```
Comment (danielballan) [inline chxtools/chx_wrapper.py:47]: `caget` shouldn't be used in a plan. @mrakitin Can you open an issue for this too?

File: chxtools/chx_wrapper.py [hunk 7]
```diff
@@ -43,30 +44,32 @@ def cw_ascan(mot,xmin,xmax,npoints,acqt='default',pos_ret=True):
...
+        except:
+            print('could not set exposure time to {}'.format(acqt))
+
     # execute the scan
     ascan(mot,xmin,xmax,npoints)
-    
+
     # put beamline back into initial state
-    if pos_ret==True:
+    if pos_ret is True:
```
Comment (danielballan) [inline chxtools/chx_wrapper.py:64]: `if pos_ret:` is sufficient here.

File: chxtools/chx_wrapper.py [hunk 8]
```diff
@@ -78,25 +81,27 @@ def cw_dscan(mot,mdx,pdx,npoints,acqt='default',pos_ret=True):
     WILL NOT WORK FOR A LIST OF DETECTORS!
     """
     # current detector:
-    acq_pv=session_mgr[ascan.user_detectors[0].name].pvname
-    
+    acq_pv = session_mgr[ascan.user_detectors[0].name].pvname
```
Comment (danielballan) [inline chxtools/chx_wrapper.py:84]: `session_mgr` is long-gone too. I guess we should just make a note to revisit this entire file. I suspect it hasn't been used in years, so we need to decide whether to delete it or overhaul it.

**Comment by yugangzhang** [comment]
Great Job!

**Comment by danielballan** [comment]
@yugangzhang There were some comments that we had planned to address before merging here.

**Comment by yugangzhang** [comment]
@danielballan I think there might be some issues with handler, but it's in eiger_io package. I will revert it.

File: chxtools/handlers2.py (hunk 2 above)
Comment (mrakitin) [reply chxtools/handlers2.py:105]: Done!

**Comment by mrakitin** [comment]
I implemented suggestions from @jrmlhermitte and @danielballan, see https://github.com/NSLS-II-CHX/chxtools/pull/18.

**Pull request merged**